lib/susedistribution.pm

```diff
@@ -339,9 +339,17 @@ sub ensure_installed {
         record_soft_failure "boo#1091353 - pkcon doesn't find existing pkg - falling back to zypper";
     }
     elsif ($ret =~ /pkcon-status-5/) {
-        record_info 'pkcon failed', 'Return value meaning: "Nothing useful was done", trying fallback to zypper"';
-        $self->_ensure_installed_zypper_fallback($pkglist);
-        record_soft_failure 'boo#1100134 - pkcon randomly fails to download packages';
+        if (!check_screen('pkcon-pkg-already-installed')) {
+            record_info 'pkcon failed', 'Return value meaning: "Nothing useful was done", trying fallback to zypper"';
+            $self->_ensure_installed_zypper_fallback($pkglist);
```

Comment:
I wonder if this is actually needed, I mean, if pkcon failed, nothing useful was done.... what's the point?... @asdil12 can you try and test with a package that doesn't exist?

Reply:
There are usecases - zypper is more reliable in general - also it can make use of provides which pkcon cant.

`pkcon install unrar` will fail as there is no such pkg but `zypper in unrar` works and installs `unrar_wrapper` which provides `unrar`.